### PR TITLE
 expand RUST AES-CBC coverage for padding, truncation, buffer sizing, and streaming

### PR DESCRIPTION
**Pull request by meilu-git**
[test] expand  RUST AES-CBC coverage for padding, truncation, buffer sizing, and streaming

### CODE REVIEW

**Comment by meilu-git** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: api/tests/src/algo/aes/cbc_tests.rs [hunk 1]
```diff
@@ -430,3 +597,439 @@ fn test_cbc_streaming_encrypt_streaming_decrypt_no_pad_128_diff_boundaries(sessi
...
+    // Encrypt one block
+    let out_len = enc_ctx.update(&pt, None).unwrap();
+    let mut out = vec![0u8; out_len];
+    let _written = enc_ctx.update(&pt, Some(&mut out)).unwrap();
+    // assert!(written > 0);
+
+    // "Operation mismatch" is treated as no-op by backend
+    let result = enc_ctx.update(&out, None).unwrap();
+    assert_eq!(result, 0);
+}
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply api/tests/src/algo/aes/cbc_tests.rs]: These sweeps were originally added to stress padding edge cases, but the full 0..=128 range is heavier than needed for CI/emulator runs.

I’ve reduced them to a boundary-focused set of representative lengths (0, 1, 15, 16, 17, 31, 32, 33, 127, 128) to preserve coverage while significantly reducing runtime.

File: api/tests/src/algo/aes/cbc_tests.rs [hunk 2]
```diff
@@ -181,6 +186,189 @@ fn run_cbc_roundtrip(
...
+    let pt = vec![0x11; AES_CBC_BLOCK_SIZE + 5];
+    let ct = cbc_encrypt(&key, true, &iv, &pt).unwrap();
+
+    // PKCS#7 padding length = 11 (0x0B)
+    let pad_len = AES_CBC_BLOCK_SIZE - (pt.len() % AES_CBC_BLOCK_SIZE);
+
+    for i in 0..pad_len {
+        let mut bad = ct.clone();
+        let idx = bad.len() - 1 - i; // only corrupt padding bytes
+        bad[idx] ^= 0xAA;
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply api/tests/src/algo/aes/cbc_tests.rs]: I haven't seen it failed, will keep an eye out.

File: api/tests/Cargo.toml [hunk 3]
```diff
@@ -15,6 +15,7 @@ cmake.workspace = true
 azihsm_api.workspace = true
 azihsm_api_tests_macro.workspace = true
 azihsm_crypto.workspace = true
+getrandom = "0.2"
```
Comment (mhatrevi) [inline api/tests/Cargo.toml]: We already have rand capability in our crypto. Can we use that instead of adding a new crate dependency?

**Review by rajesh-gali — APPROVED** [review]
